### PR TITLE
Feature - New grid component

### LINKED ISSUE #503: Bento grid component
I would like to add a new Bento Grid layout to the Grid section of HyperUI. This layout will provide a unique way to showcase content using variable-sized grid items, making it visually appealing and flexible for use cases such as portfolios and gallery sections.

**Feature Details:**
- Each grid item will have dynamic sizing and placement, similar to the Bento Box style.
- The grid will support responsive layouts and hover animations.
- I will add a component preview and documentation to explain how to implement and use it.

**Why this feature?**
The Bento Grid layout offers a modern, Pinterest-like grid that could enhance the variety of layouts available in HyperUI.

If this addition is accepted, I will implement it and submit a pull request. Please let me know if there are any specific requirements or suggestions for this component! @markmead 

"Here’s a rough wireframe of the Bento Grid layout I plan to implement. The grid items are of varying sizes, and the layout supports responsive design."
image

### PR DESCRIPTION
**Pull request by mrpankajpandey**
Fixed #503
Description
This PR adds a Bento Grid layout to the Grid section of HyperUI. The Bento Grid allows for creating layouts with variable-sized grid items, making it perfect for showcasing content like portfolios or galleries.

Key Features
 1. Dynamic grid items with flexible sizing.
 2. Responsive design suitable for different screen sizes.
 3. Hover animations for interactive experience.

Screenshots

image

### CODE REVIEW

**Review by markmead — COMMENTED** [review]
Shouldn't have changes to `package-lock.json` or `yarn.lock`.

---

I'm not sure I understand how this is dynamic, looks like you have to set the row/col values yourself. I was thinking it would be more based on the content, kinda like how it's done here - https://www.hyperui.dev/components/marketing/testimonials#component-5

**Comment by mrpankajpandey** [comment]
Thank you for your feedback! @markmead 

I appreciate your insights on the dynamic nature of the Bento Grid. I initially implemented it with manual row and column settings to provide users with more layout control. However, I understand the importance of making it more content-driven, similar to the testimonials component. I can definitely work on adjusting the implementation to better align with that approach.

Additionally, would it be acceptable to move the Bento Grid component to the testimonials section? I believe it could fit well there and enhance the overall design.

**Comment by markmead** [comment]
I might be mistaken but I thought a bento grid was dynamic which, if that's the case, I think it needs to be dynamic on here as well. We can use some dummy data to make that happen. No need to move it to testimonials, it should stay in grid.

**Comment by mrpankajpandey** [comment]
Got it! I’ll make the grid dynamic using some dummy data, similar to how other components are structured. I’ll ensure it adapts based on content instead of using manual row/column settings.

Can I use placeholder text and images like the previous grid components, [Link](https://github.com/markmead/hyperui/issues/503#issue-2563167677) or would you prefer any specific data for the demo?

File: public/components/application-ui-grids/13.html [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<div id="bento-grid" class="container">
```
Comment (markmead) [inline public/components/application-ui-grids/13.html:1]: No need for `id` and `container` isn't used on HyperUI.

File: public/components/application-ui-grids/13.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  <h2 class="h2-style"></h2>
+  <div id="grid-container" class="flex flex-col gap-6 lg:grid lg:grid-cols-3">
```
Comment (markmead) [inline public/components/application-ui-grids/13.html:3]: No need for `id`.

File: public/components/application-ui-grids/13.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    <div
+      class="group row-start-1 row-end-3 rounded-2xl bg-slate-100 p-[1px] hover:bg-gradient-to-br hover:from-red-200 hover:via-purple-200 hover:to-yellow-200"
+    >
+      <div
+        class="flex h-full w-full flex-col items-center gap-6 rounded-2xl bg-slate-50 p-6 group-hover:bg-gradient-to-br group-hover:from-red-50 group-hover:via-purple-50 group-hover:to-yellow-50"
+      >
+        <h3 class="text-2xl">Native Notifications</h3>
+        <img src="https://github.com/user-attachments/assets/06db2547-c6de-4765-8638-96114a46b4f7" alt="" />
```
Comment (markmead) [inline public/components/application-ui-grids/13.html:11]: Can't use `github.com/user-attachments...` for a `src`.

**Review by markmead — COMMENTED** [review]
Not convinced this is truly dynamic, still looks to be using fixed sizes which I'm not interested in to be honest. I'd rather see something like that added in the same style as the existing HyperUI grid components.

**Pull request closed without merging**